interfaces/cython/SConscript

```diff
@@ -104,9 +106,238 @@ else:
+                    ) from None
+                else:
+                    raise  # pragma: no cover
+        self._phase.${name} = ${n0}.magnitude, ${n1}.magnitude
+""")
+
+tp_setter_2_properties = ["TP", "DP", "HP", "SP", "SV", "TD", "UV"]
+pf_setter_2_properties = ["PQ", "TQ", "PV", "SH", "ST", "TH", "TV", "UP", "VH"]
+
+thermophase_2_setters = []
+for name in tp_setter_2_properties:
+    d = dict(name=name, n0=SYMBOL[name[0]], u0=UNITS[name[0]], n1=SYMBOL[name[1]],
+             u1=UNITS[name[1]])
+    thermophase_2_setters.append(setter_2_template.substitute(d))
+
+purefluid_2_setters = []
+for name in pf_setter_2_properties:
+    d = dict(name=name, n0=SYMBOL[name[0]], u0=UNITS[name[0]], n1=SYMBOL[name[1]],
+             u1=UNITS[name[1]])
+    purefluid_2_setters.append(setter_2_template.substitute(d))
+
+setter_3_template = Template("""
+    @property
+    @copy_doc
+    def ${name}(self):
+        ${n0}, ${n1}, ${n2} = self._phase.${name}
+        return Q_(${n0}, ${u0}), Q_(${n1}, ${u1}), Q_(${n2}, ${u2})
+
+    @${name}.setter
+    def ${name}(self, value):
+        ${n0} = value[0] if value[0] is not None else self.${n0}
+        ${n1} = value[1] if value[1] is not None else self.${n1}
+        for val, unit in ((${n0}, ${u0}), (${n1}, ${u1})):
+            try:
+                val.ito(unit)
+            except AttributeError as e:
+                if "'ito'" in str(e):
+                    raise CanteraError(
+                        f"Value {val!r} must be an instance of a pint.Quantity class"
+                    ) from None
+                else:
+                    raise  # pragma: no cover
+        if value[2] is not None:
+            try:
+                ${n2} = value[2].to(${u2}).magnitude
+            except AttributeError:
+                ${n2} = value[2]
+        else:
+            ${n2} = self.${n2}.magnitude
+        self._phase.${name} = ${n0}.magnitude, ${n1}.magnitude, ${n2}
+""")
+
+tp_setter_3_properties = [
+    "TPX", "TPY", "DPX", "DPY", "HPX", "HPY", "SPX", "SPY", "SVX", "SVY", "TDX", "TDY",
+    "UVX", "UVY"
+]
+
+thermophase_3_setters = []
+for name in tp_setter_3_properties:
+    d = dict(name=name, n0=SYMBOL[name[0]], u0=UNITS[name[0]], n1=SYMBOL[name[1]],
+             u1=UNITS[name[1]], n2=SYMBOL[name[2]], u2=UNITS[name[2]])
+    thermophase_3_setters.append(setter_3_template.substitute(d))
+
+getter_3_template = Template("""
+    @property
+    @copy_doc
+    def ${name}(self):
+        ${n0}, ${n1}, ${n2} = self._phase.${name}
+        return Q_(${n0}, ${u0}), Q_(${n1}, ${u1}), Q_(${n2}, ${u2})
+""")
+
+pf_getter_3_properties = ["DPQ", "HPQ", "SPQ", "SVQ", "TDQ", "UVQ"]
+
+purefluid_3_getters = []
+for name in pf_getter_3_properties:
+    d = dict(name=name, n0=name[0], u0=UNITS[name[0]], n1=name[1], u1=UNITS[name[1]],
+             n2=name[2], u2=UNITS[name[2]])
+    purefluid_3_getters.append(getter_3_template.substitute(d))
```

Comment:
I didn't want to use `f` strings because I'd have to do like
```python
for name in ....:
    list.append(f"""long string""")
```
which seems even clunkier. Using `.format()` would also require constructing the dict of substitutions, or else pass them as kwargs to the `.format()` call, which amounts to the same thing. There's no particular reason to prefer `string.Template` here, it just seemed easier and purpose built.